search_blackbox_test.go

```diff
@@ -200,3 +210,53 @@ func TestUnwantedCharactersRelatedToSearchLogic(t *testing.T) {
 	_, sr := test.ShowSearchOK(t, nil, nil, controller, nil, nil, q)
 	assert.Equal(t, 0, len(sr.Data))
 }
+
+// TestAutoRegisterHostURL copies much of the codebase from search_testing.go->ShowSearchOK
+// and customises the values to add custom Host in the call.
+// This test checks if client's host is neatly registered as a KnwonURL or not
+func TestAutoRegisterHostURL(t *testing.T) {
+	resource.Require(t, resource.Database)
+	var resp interface{}
+	var respSetter goatest.ResponseSetterFunc = func(r interface{}) { resp = r }
+	service := getServiceAsUser()
+	newEncoder := func(io.Writer) goa.Encoder { return respSetter }
+	service.Encoder = goa.NewHTTPEncoder() // Make sure the code ends up using this decoder
+	service.Encoder.Register(newEncoder, "*/*")
+	rw := httptest.NewRecorder()
+	query := url.Values{}
+	customHost := "curious.unknown.domain"
+	u := &url.URL{
+		Path:     fmt.Sprintf("/api/search"),
+		RawQuery: query.Encode(),
+		Host:     customHost,
+	}
+	req, err := http.NewRequest("GET", u.String(), nil)
+	if err != nil {
+		panic("invalid test " + err.Error()) // bug
+	}
+	prms := url.Values{}
+	prms["q"] = []string{`http://any.url:8080/different-path/`} // any value will do
```

Comment:
Wouldn't it make more sense to search for something using the customHost and actually finding the ID, as oppose to just checking if customHost was registered as known?